### ISSUE TITLE
EnumerationGraphType.Coerce compares Value instead of Name

### ISSUE THREAD

**Issue by JarnoNijboer** [graphql-dotnet/graphql-dotnet#79]
#### Problem
It seems that graphql-dotnet resolves the value of a ```EnumerationGraphType``` differently then ```graphql-js``` and ```graphql-java``` implementations.

I have query with has a variable of type ```InputObjectGraphType```, which contains a field of type ```EnumerationGraphType```.
When using the enum below, graphql-dotnet expects the value ```0, 1, 2```, where as GraphiQL forces me to use one of the names ```RED, GREEN, BLUE```.
After reading the [spec](https://github.com/facebook/graphql/blob/master/spec/Section%203%20--%20Type%20System.md#enums), and checking the [graphql-js](https://github.com/graphql/graphql-js/blob/master/src/type/definition.js#L734) reference and [java](https://github.com/graphql-java/graphql-java/blob/master/src/main/java/graphql/schema/GraphQLEnumType.java#L27) implementation, it seems that graphql-dot-net doesn't follow the spec/reference implementation. 
It compares the given value with ```EnumValue.Value``` instead of ```EnumValue.Name```. It returns also the ```EnumValue.Name``` instead of ```EnumValue.Value```.

##### Example of Enum:
```csharp
public class RGBType : EnumerationGraphType
{
    public RGBType()
    {
        Name = "RGBType";

        AddValue("RED", string.Empty, 0);
        AddValue("GREEN", string.Empty, 1);
        AddValue("BLUE", string.Empty, 3);
    }
}
```

#### Solution
Replacing the contents of the [EnumerationGraphType.Coerce](https://github.com/graphql-dotnet/graphql-dotnet/blob/master/src/GraphQL/Types/EnumerationGraphType.cs#L33)  method, with the following code, alings the behavior of ```graphql-dotnet``` with the ```graphql-js``` reference implementation.
```csharp
public override object Coerce(object value)
{
    var found = Values.FirstOrDefault(v => v.Name.Equals(value));
    return found != null ? found.Value : null;
}
```

**Comment by JarnoNijboer** [comment]
My solution works for correct for validation, but fails when ```CoerceValue``` method from ```DocumentExecutor``` is executed.
The solution that @HasAndries mentioned in [#70](https://github.com/graphql-dotnet/graphql-dotnet/pull/70#issuecomment-217399867) is working perfectly for me

**Comment by joemcbride** [comment]
I recall looking at this previously now.  One of the issues is how the `DocumentExecutor` is trying to validate Enumerations.  That solution will "work", though it hides that issue.  I'm working on fixing that.

**Comment by joemcbride** [comment]
This should be fixed in [v0.8.0](https://github.com/graphql-dotnet/graphql-dotnet/releases/tag/v0.8.0) - please re-open if you find an issue.
